internal/helm/helm.go

```diff
@@ -412,6 +412,16 @@ func upgradeRelease(ctx context.Context, local client.Client, restClientGetter g
 	return err
 }
 
+// RenderChartManifests renders a Helm chart with the given values and returns the rendered Kubernetes manifests.
+// This performs a dry-run installation to get the manifests without actually deploying anything.
+func RenderChartManifests(ctx context.Context, local client.Client, restClientGetter genericclioptions.RESTClientGetter, pluginDefinitionSpec greenhousev1alpha1.PluginDefinitionSpec, plugin *greenhousev1alpha1.Plugin) (string, error) {
+	helmRelease, err := installRelease(ctx, local, restClientGetter, pluginDefinitionSpec, plugin, true)
```

Comment:
Besides the return value this is identical to `TemplateHelmChartFromPlugin`. 
```suggestion
	helmRelease, err := TemplateHelmChartFromPlugin(ctx, local, restClientGetter, pluginDefinitionSpec, plugin)
```
or directly use `TemplateHelmChartFromPlugin` and access the manifest from the release, when passing them to `createRegistryMirrorPostRenderer`

`postRenderer := createRegistryMirrorPostRenderer(mirrorConfig, release.Manifest)`